compiler/luci-interpreter/src/core/RuntimeGraph.h

```diff
@@ -46,13 +46,18 @@ class RuntimeGraph
   void execute() const;
 
 private:
+  // Build tensor deallocation plan
+  void prepareDeallocPlan() const;
+
   RuntimeModule *_owning_module;
   std::vector<std::unique_ptr<Tensor>> _tensors;
   std::vector<Tensor *> _input_tensors;
   std::vector<Tensor *> _output_tensors;
 
   // Kernels in execution order.
   std::vector<std::unique_ptr<Kernel>> _kernels;
+  // Tensors that are not used anymore after given op
+  mutable std::vector<std::vector<Tensor *>> _tensor_dealloc_plan;
```

Comment:
I propose to add some non-constant method, like "prepare" and do allocation/deallocation planning there, so we do not need mutable magic here.

Other way is to add incremental planning in addKernel method, but I think this will complicate code a little.

Reply:
`mutable` usage is typical for storing cached data, like here. This cache does not keep any additional state, it duplicates the information already available in other fields. One way to hide `mutable` is using `unique_ptr` indirection, interpreter uses such an approach many timer (like, say, keeping temporary tensors). Adding `prepare` non-const method would complicate the interface if the class.